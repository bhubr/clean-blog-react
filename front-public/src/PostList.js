import React from 'react'
import PropTypes from 'prop-types'
import Post from './Post'

class PostList extends React.Component {
  render () {
    return (
      <div>
        {
          this.props.posts.map((post, i) => <Post key={i} id={ post.id } title={ post.title } content={ post.content } />)
        }
        <div className="clearfix">
          <a className="btn btn-primary float-right" href="#">Older Posts &rarr;</a>
        </div>
      </div>
    )
  }
}

PostList.propTypes = {
  posts: PropTypes.array
}

export default PostList
