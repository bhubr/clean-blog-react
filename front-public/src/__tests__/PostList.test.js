// PostList.test.js
import React from 'react'
import PostList from '../PostList'
import renderer from 'react-test-renderer'

const posts = [
  { title: 'Test post #1', content: 'Some content' },
  { title: 'Test post #2', content: 'Some more content' }
]

test('Check that output is correct', () => {
  // Render component
  const component = renderer.create(
    <PostList posts={posts} />,
  )
  // Convert component to a tree of objects
  let tree = component.toJSON()
  // Match against snapshot
  expect(tree).toMatchSnapshot()
})
