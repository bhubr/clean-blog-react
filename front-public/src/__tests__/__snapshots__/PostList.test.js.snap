// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that output is correct 1`] = `
<div>
  <div>
    <h2>
      Test post #1
    </h2>
    <div>
      <p>
        Some content
      </p>
    </div>
  </div>
  <div>
    <h2>
      Test post #2
    </h2>
    <div>
      <p>
        Some more content
      </p>
    </div>
  </div>
</div>
`;
